SDL/Makefile

```diff
@@ -32,21 +30,31 @@ MEMFLAG_FLAGS        =
 CACHEFLAG_FLAGS      = -DCACHE_ALLOC
 T5CUTFLAGS           = $(T5DNNFLAG) $(T5RZCHI2FLAG) $(T5RPHICHI2FLAG)
 
-#
-# how to make it 
-#
+ifeq ($(BACKEND), cpu)
+  LD                 = g++
+  SOFLAGS            = -g -shared -fPIC
+  ALPAKABACKEND      = $(ALPAKASERIAL)
+  COMPILE_CMD        = $(LD) -c -O2
```

Comment:
in mkfit we have `-Ofast -fno-reciprocal-math -fopenmp-simd`
it's worth exploring; can be done after this PR

Reply:
Here's the timing with the current flags:
```
Total Timing Summary
Average time for map loading = 2601.87 ms
Average time for input loading = 19196.7 ms
Average time for SDL::Event creation = 0.891095 ms
   Evt    Hits       MD       LS      T3       T5       pLS       pT5      pT3      TC       Reset    Event     Short             Rate
   avg     57.7    417.4    142.5     85.1    126.2    601.5    743.1    154.6    111.2     14.4    2453.9    1794.7+/- 449.0    2458.5   explicit[s=1]
   avg     55.5    412.9    134.4     77.4    120.0    613.0    762.6    160.2    114.0      1.8    2451.8    1783.3+/- 467.9     620.1   explicit[s=4]
   avg    116.9    459.1    169.7     97.0    136.0    663.8    823.0    174.2    122.7      1.7    2764.1    1983.3+/- 518.1     198.5   explicit[s=16]
   avg    407.9    564.3    262.1    154.7    160.7    731.2    910.0    191.9    133.9      2.3    3518.9    2379.8+/- 632.9     142.3   explicit[s=32]
   avg   2046.2    998.8    736.1    471.3    291.6   1123.1   1253.2    288.5    190.9      3.6    7403.4    4234.1+/- 1487.9     148.4   explicit[s=64]
   avg     52.8    409.7    139.9     82.0    123.6    608.2    463.6    151.6    114.7      0.6    2146.6    1485.6+/- 366.6    2159.9   explicit_cache[s=1]
   avg     54.9    409.3    132.7     72.8    118.2    628.0    486.2    158.6    118.3      1.0    2180.0    1497.1+/- 387.6     560.8   explicit_cache[s=4]
   avg    123.8    461.4    172.2     99.1    138.1    678.8    532.6    173.9    128.9      0.9    2509.5    1707.0+/- 441.2     179.8   explicit_cache[s=16]
   avg    445.7    552.4    269.4    155.3    166.2    764.4    586.5    190.6    142.8      1.2    3274.5    2064.4+/- 541.0     129.7   explicit_cache[s=32]
   avg   2223.9    991.9    766.6    470.0    278.3   1103.8    855.8    280.3    198.1      1.5    7170.2    3842.5+/- 1451.5     566.9   explicit_cache[s=64]
```
And here's the timing with your proposed flags:
```
Total Timing Summary
Average time for map loading = 2560.93 ms
Average time for input loading = 19243.5 ms
Average time for SDL::Event creation = 0.843377 ms
   Evt    Hits       MD       LS      T3       T5       pLS       pT5      pT3      TC       Reset    Event     Short             Rate
   avg     52.3    292.0    129.0     74.7     97.1    925.4     94.9    115.8    126.2      9.8    1917.2     939.5+/- 218.2    1921.5   explicit[s=1]
   avg     54.5    286.2    121.2     67.0     91.1    958.7     98.5    121.1    130.5      1.4    1930.2     917.1+/- 231.5     484.7   explicit[s=4]
   avg    143.6    346.7    175.6     98.7    109.7   1032.3    114.8    135.4    139.6      1.4    2297.8    1121.9+/- 303.2     165.0   explicit[s=16]
   avg    737.4    772.9    417.5    241.7    216.3   1140.2    144.6    276.7    151.4      2.4    4101.1    2223.5+/- 778.9     156.4   explicit[s=32]
   avg   3534.9   2031.1   1075.8    658.3    514.9   1573.3    232.5    653.2    208.1      2.8   10485.1    5376.8+/- 2131.0     190.0   explicit[s=64]
   avg     52.4    293.8    130.1     76.1     97.8    929.4     97.1    117.0    128.1      0.6    1922.4     940.6+/- 219.4    1936.6   explicit_cache[s=1]
   avg     55.2    292.8    125.5     68.5     93.0    958.3    103.4    122.6    132.4      0.7    1952.5     939.0+/- 235.4     496.5   explicit_cache[s=4]
   avg    151.0    363.5    177.6    100.5    112.7    987.5    116.2    135.7    138.1      0.9    2283.7    1145.2+/- 327.2     162.8   explicit_cache[s=16]
   avg    741.8   1289.9    422.9    242.6    263.4   1030.5    141.3    300.8    143.8      1.4    4578.4    2806.1+/- 1056.0     171.9   explicit_cache[s=32]
   avg   3121.9   2896.8    999.2    622.9    597.4   1419.9    221.9    721.0    191.2      2.5   10794.6    6252.9+/- 2510.1     200.6   explicit_cache[s=64]
```
It's interesting that it's kind of a mixed bag. Overall, the timing is better (especially the huge improvement in the pT5 step), but the pLS and TC steps are a bit slower. Also, for 64 streams it is significantly slower.

Apart from that, I also tested optimization flags for the GPU version. As far as I can find, the device code is O3 optimized by default, but the host code is not. So I tried adding `-O3` (`nvcc` doesn't recognize `-Ofast`) to optimize host code and it makes a noticeable difference. Here is the timing with the current flags:
```
Total Timing Summary
Average time for map loading = 7779.93 ms
Average time for input loading = 19089.7 ms
Average time for SDL::Event creation = 0.0890323 ms
   Evt    Hits       MD       LS      T3       T5       pLS       pT5      pT3      TC       Reset    Event     Short             Rate
   avg      9.1      2.2      0.4      2.8      3.4      1.0      1.5      1.2      1.2      6.9      29.6      19.6+/-  2.2      33.3   explicit[s=1]
   avg     17.1      2.5      0.5      4.8      5.1      1.0      3.8      3.2      1.6     10.8      50.3      32.2+/-  6.0      30.2   explicit[s=2]
   avg     40.4      2.7      0.6      9.5     11.8      1.2      9.2      7.3      2.0     25.0     109.6      68.0+/- 15.3      30.1   explicit[s=4]
   avg     70.0      2.7      0.5     15.1     18.1      1.1     15.0     11.0      1.9     48.1     183.5     112.4+/- 26.9      32.7   explicit[s=6]
   avg     90.2      2.7      0.5     18.5     23.5      1.1     19.7     15.9      1.9     71.5     245.4     154.2+/- 63.9      34.0   explicit[s=8]
   avg      5.1      2.2      0.4      1.9      2.0      1.0      1.4      1.1      1.3      0.5      16.7      10.6+/-  2.2      21.1   explicit_cache[s=1]
   avg      6.7      2.4      0.4      2.6      2.5      1.0      2.0      1.5      1.7      0.9      21.7      14.0+/-  3.5      17.5   explicit_cache[s=2]
   avg     11.7      3.1      0.8      4.4      4.0      1.1      3.4      2.2      2.8      1.3      34.8      21.9+/-  7.2      11.5   explicit_cache[s=4]
   avg     18.2      3.9      1.3      6.0      5.9      1.3      4.8      3.4      4.1      2.3      51.1      31.6+/-  9.3      10.2   explicit_cache[s=6]
   avg     25.6      4.9      1.6      7.4      7.9      1.5      6.5      4.2      4.5      3.6      67.7      40.6+/- 14.7      10.3   explicit_cache[s=8]
```
And here's the timing with `-O3`:
```
Total Timing Summary
Average time for map loading = 3751.16 ms
Average time for input loading = 19153.7 ms
Average time for SDL::Event creation = 0.0842753 ms
   Evt    Hits       MD       LS      T3       T5       pLS       pT5      pT3      TC       Reset    Event     Short             Rate
   avg      8.8      2.2      0.3      2.7      3.3      1.0      1.2      0.9      1.2      6.8      28.3      18.5+/-  2.4      32.2   explicit[s=1]
   avg     17.5      2.6      0.4      4.8      4.7      1.0      3.5      2.8      1.7     11.4      50.3      31.9+/-  6.8      29.8   explicit[s=2]
   avg     33.2      2.7      0.5     10.2      8.9      1.1     10.1      7.6      1.9     26.3     102.5      68.2+/- 20.0      28.8   explicit[s=4]
   avg     51.5      2.7      0.6     15.0     12.9      1.1     18.7     14.1      1.8     43.1     161.6     108.9+/- 40.0      28.9   explicit[s=6]
   avg     67.5      2.7      0.6     21.4     15.8      1.2     28.9     19.8      1.8     64.1     223.7     155.1+/- 46.8      29.3   explicit[s=8]
   avg      4.4      2.2      0.3      1.7      1.8      1.0      0.9      0.7      1.2      0.1      14.4       9.1+/-  2.0      18.4   explicit_cache[s=1]
   avg      6.1      2.5      0.5      2.6      2.5      1.0      1.6      1.1      1.9      0.3      19.9      12.8+/-  3.3      12.4   explicit_cache[s=2]
   avg     10.9      3.1      0.8      4.3      3.7      1.1      3.0      1.8      3.2      0.5      32.4      20.4+/-  5.3       9.8   explicit_cache[s=4]
   avg     15.7      3.8      1.3      5.6      5.5      1.2      4.1      2.5      4.2      0.8      44.6      27.7+/- 11.1       9.8   explicit_cache[s=6]
   avg     21.2      4.7      1.7      7.4      7.3      1.4      5.7      3.3      5.9      0.9      59.6      36.9+/- 12.3       8.9   explicit_cache[s=8]
```
It's a small improvement, but it's probably worth it.

Reply:
some changes are a bit too large. 
Are the results the same for physics?

May I ask you to post screenshots instead of plain text: plain text tables do not fit in the browser mini-window and it takes an effort to compare

Reply:
> some changes are a bit too large.

I'm more concerned about slower cases. It seems to be just in pLS and TC. Could they be both related to the "checkHitsPLs"?

Reply:
That's a good point. I'll focus on the checkHitspLS issues first. And okay, I'll use screenshots from now on. Should we merge this PR and move this discussion somewhere else since it's kind of unrelated?

Reply:
pT5 743 -> 95 also seems like suspiciously large change

Reply:
I agree with the comment that we should move this to a different PR. Optimizing the CPU backend is a bit out of the scope of this PR, I think.